### PR TITLE
Initial commit to fix IParser::Pos.max_depth issue

### LINKED ISSUE #6681: Hardcoded IParser.h max_depth = 1000 should be configurable.
This pull request https://github.com/yandex/ClickHouse/commit/0a9787c7f3eb295ed94c7a6d35fe702ad59648e3#diff-7ecb930104742be8bbaf5b53aa1f00deR60
 
`uint32_t max_depth = 1000;`

broke some existing working SQL.

https://t.me/clickhouse_ru/115334

$ wc amp_task.sql
      60    4122   57661 amp_task.sql

### PR DESCRIPTION
**Pull request by qMBQx8GH**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):

New setting introduced: max_parser_depth. To control maximum stack size and allow long complex queries. This fixes #6681 and #7668.

### CODE REVIEW

File: dbms/src/Databases/DatabaseOrdinary.cpp [hunk 1]
```diff
@@ -122,12 +122,12 @@ void DatabaseOrdinary::loadStoredObjects(
     FileNames file_names;
 
     size_t total_dictionaries = 0;
-    iterateMetadataFiles(context, [&file_names, &total_dictionaries, this](const String & file_name)
+    iterateMetadataFiles(context, [context, &file_names, &total_dictionaries, this](const String & file_name)
```
Comment (alexey-milovidov) [inline dbms/src/Databases/DatabaseOrdinary.cpp]: Maybe &context to avoid copy?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Ok. And we also need a test (simple .sql test is enough, see `dbms/tests/queries`) to check that the setting works.

File: dbms/src/Databases/DatabaseOrdinary.cpp (hunk 1 above)
Comment (qMBQx8GH) [reply dbms/src/Databases/DatabaseOrdinary.cpp]: ok. I've never used C++ lambda functions.

**Comment by qMBQx8GH** [comment]
> Ok. And we also need a test (simple .sql test is enough, see `dbms/tests/queries`) to check that the setting works.

done

**Review by alexey-milovidov — APPROVED** [review]
Ok.

**Pull request merged**